### PR TITLE
fixing getopts, adding host specific files to backup, refactoring

### PR DESCRIPTION
**Pull request by sp3nx0r**
Update script had an error in the restore (due to getopts flag typos). Also added specific files that are configured during build process (/etc/hosts, /etc/hostname, /etc/motd) to keep the critter's name from being overwritten. Also refactored execution of steps to make a tad more sense.

### CODE REVIEW

File: scripts/update_pwnagotchi.sh [hunk 1]
```diff
@@ -36,70 +37,79 @@ function test_root() {
...
 done
 shift $((OPTIND-1))
 
+echo "[+] Checking prerequisites."
+test_root
+test_github
+
 # clean up old files, clone master, set checkout to commit if needed.
 echo "[+] Cloning to $GIT_FOLDER..."
-rm $GIT_FOLDER -rf
```
Comment (justin-p) [inline scripts/update_pwnagotchi.sh]: I added this here before the clone to ensure git would not throw a fit about the path being in use.

Comment (sp3nx0r) [reply scripts/update_pwnagotchi.sh]: doh, missed that from debugging. fixed in latest push

**Comment by sp3nx0r** [comment]
Added additional missing file as pointed out in PR#71

**Comment by evilsocket** [comment]
conflicts

**Comment by sp3nx0r** [comment]
Resolved merge conflict. This PR now fixes the getopts issue, adds missing /etc/motd from @incredincomp's PR, and tweaks echo statements for clarity

**Pull request merged**